packages/module/src/AttachMenu/Dropdown.tsx

```diff
@@ -0,0 +1,195 @@
+  maxMenuHeight,
+  shouldFocusFirstItemOnOpen = true,
+  ...props
+}: DropdownProps) => {
+  const localMenuRef = React.useRef<HTMLDivElement>();
+  const localToggleRef = React.useRef<HTMLButtonElement>();
+  const ouiaProps = useOUIAProps(Dropdown.displayName ?? 'Dropdown', ouiaId, ouiaSafe);
+
+  const menuRef = (innerRef as React.RefObject<HTMLDivElement>) || localMenuRef;
+  const toggleRef =
+    typeof toggle === 'function' || (typeof toggle !== 'function' && !toggle.toggleRef)
+      ? localToggleRef
+      : (toggle?.toggleRef as React.RefObject<HTMLButtonElement>);
+
+  const prevIsOpen = React.useRef<boolean>(isOpen);
+  React.useEffect(() => {
+    // menu was opened, focus on first menu item
+    if (prevIsOpen.current === false && isOpen === true && shouldFocusFirstItemOnOpen) {
+      setTimeout(() => {
+        const firstElement = menuRef?.current?.querySelector(
+          'li button:not(:disabled),li input:not(:disabled),li a:not([aria-disabled="true"])'
+        );
+        firstElement && (firstElement as HTMLElement).focus();
+      }, 0);
+    }
+
+    prevIsOpen.current = isOpen;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen]);
+
+  React.useEffect(() => {
+    const handleMenuKeys = (event: KeyboardEvent) => {
+      // Close the menu on tab or escape if onOpenChange is provided
+      if (
+        isOpen &&
+        onOpenChange &&
+        (menuRef.current?.contains(event.target as Node) || toggleRef.current?.contains(event.target as Node))
+      ) {
+        if (onOpenChangeKeys.includes(event.key)) {
+          onOpenChange(false);
+          toggleRef.current?.focus();
+        }
+      }
+    };
+
+    const handleClick = (event: MouseEvent) => {
+      // If the event is not on the toggle and onOpenChange callback is provided, close the menu
+      if (isOpen && onOpenChange && !toggleRef?.current?.contains(event.target as Node)) {
+        if (isOpen && !menuRef.current?.contains(event.target as Node)) {
+          onOpenChange(false);
+        }
+      }
+    };
+
+    window.addEventListener('keydown', handleMenuKeys);
+    window.addEventListener('click', handleClick);
+
+    return () => {
+      window.removeEventListener('keydown', handleMenuKeys);
+      window.removeEventListener('click', handleClick);
+    };
+  }, [isOpen, menuRef, toggleRef, onOpenChange, onOpenChangeKeys]);
+
+  const scrollable = maxMenuHeight !== undefined || menuHeight !== undefined || isScrollable;
+
+  const menu = (
+    <Menu
+      className={css(className)}
+      ref={menuRef}
+      onSelect={(event, value) => {
+        onSelect && onSelect(event, value);
+        shouldFocusToggleOnSelect && toggleRef.current?.focus();
+      }}
+      isPlain={isPlain}
+      isScrollable={scrollable}
+      {...props}
+      {...ouiaProps}
+    >
+      <MenuContent menuHeight={menuHeight} maxMenuHeight={maxMenuHeight}>
+        {children}
+      </MenuContent>
+    </Menu>
+  );
+  return (
+    <Popper
```

Comment:
I think you could use a PF dropdown rather than wiring one up with. Popper + menu component. Right now the tab order is a little wonky in this implementation.

Is it possible you can mimic the [kebab dropdown example](https://staging-v6.patternfly.org/components/menus/dropdown#with-kebab-toggle) to replicate this? All Menu props should be exposed to the Dropdown or DropdownList (it's been a while since I've played with it). Katie is a great person to reach out to with Dropdown questions if you can't figure it out, she has the most experience building them out 👍🏻

Reply:
Awesome! Turns out I can just pass `distance` through popperProps, no fork needed.